crates/sui-framework/sources/governance/staking_pool.move

```diff
@@ -290,6 +291,66 @@ module sui::staking_pool {
         staked_sui.delegation_activation_epoch
     }
 
+    /// Split StakedSui `self` to two parts, one with principal `split_amount`,
+    /// and the remaining principal is left in `self`.
+    /// All the other parameters of the StakedSui like `delegation_activation_epoch` or `pool_id` remain the same.
+    public fun split(self: &mut StakedSui, split_amount: u64, ctx: &mut TxContext): StakedSui {
+        StakedSui {
+            id: object::new(ctx),
+            pool_id: self.pool_id,
+            validator_address: self.validator_address,
+            delegation_activation_epoch: self.delegation_activation_epoch,
+            principal: balance::split(&mut self.principal, split_amount),
+            sui_token_lock: self.sui_token_lock,
+        }
+    }
+
+    /// Split the given StakedSui to the two parts, one with principal `split_amount`,
+    /// transfer the newly split part to the sender address.
+    public entry fun split_staked_sui(c: &mut StakedSui, split_amount: u64, ctx: &mut TxContext) {
+        transfer::transfer(split(c, split_amount, ctx), tx_context::sender(ctx));
+    }
+
+    /// Consume the staked sui `other` and add its value to `self`.
+    /// Aborts if some of the staking parameters are incompatible (pool id, delegation activation epoch, etc.)
+    public entry fun join(self: &mut StakedSui, other: StakedSui) {
```

Comment:
Nit on naming: I think the names between split/join should be consistent. Since the other one is named `split_staked_sui` this should be `join_staked_sui` (or, `split_staked_sui` should be renamed to `join`).